managed/CounterStrikeSharp.API/Modules/Menu/CenterHtmlMenu.cs

```diff
@@ -1,126 +1,68 @@
-﻿/*
- *  This file is part of CounterStrikeSharp.
- *  CounterStrikeSharp is free software: you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  CounterStrikeSharp is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with CounterStrikeSharp.  If not, see <https://www.gnu.org/licenses/>. *
- */
-
 using System.Text;
-using Microsoft.Extensions.Logging;
 
-namespace CounterStrikeSharp.API.Modules.Menu
+namespace CounterStrikeSharp.API.Modules.Menu;
+
+public class CenterHtmlMenu : BaseMenu
 {
-    public class CenterHtmlMenu : BaseMenu
+    public CenterHtmlMenu(string title) : base(title)
     {
-        public CenterHtmlMenu(string title) : base(ModifyTitle(title))
-        {
-        }
-        
-        public override ChatMenuOption AddMenuOption(string display, Action<CCSPlayerController, ChatMenuOption> onSelect, bool disabled = false)
-        {
-            var option = new ChatMenuOption(ModifyOptionDisplay(display), disabled, onSelect);
-            MenuOptions.Add(option);
-            return option;
-        }
-        
-        private static string ModifyTitle(string title)
-        {
-            if (title.Length > 32)
-            {
-                Application.Instance.Logger.LogWarning("Title should not be longer than 32 characters for a CenterHtmlMenu");
-                return title[..32];
-            }
-
-            return title;
-        }
-
-        private static string ModifyOptionDisplay(string display)
-        {
-            if (display.Length > 26)
-            {
-                Application.Instance.Logger.LogWarning("Display should not be longer than 26 characters for a CenterHtmlMenu item");
-                return display[..26];
-            }
+    }
+}
 
-            return display;
-        }
+public class CenterHtmlMenuInstance : BaseMenuInstance
+{
+    public override int NumPerPage { get; } = 4;
+    private readonly BasePlugin _plugin;
+    private string _message;
+    private Core.Listeners.OnTick _onTick;
+    public CenterHtmlMenuInstance(BasePlugin plugin, CCSPlayerController player, IMenu menu) : base(player, menu)
+    {
+        _plugin = plugin;
+        _onTick = SendMenuHtml;
+        _plugin.RegisterListener(_onTick);
     }
 
-    public class CenterHtmlMenuInstance : BaseMenuInstance
+    private void SendMenuHtml()
     {
-        private readonly BasePlugin _plugin;
-        public override int NumPerPage => 5; // one less than the actual number of items per page to avoid truncated options
+        Player.PrintToCenterHtml(_message);
```

Comment:
I like that you've moved the menu generation code outside of the OnTick function, perhaps you should call this function `DisplayCenterHtmlMenu`.